fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/QueryKeyExpression.java

```diff
@@ -258,24 +265,23 @@ private QueryKeyExpressionWithComparison nullComparison(@Nonnull Comparisons.Typ
 
     @Nonnull
     protected QueryKeyExpressionWithComparison parameterComparison(@Nonnull Comparisons.Type type, @Nonnull String param) {
-        final Function<Object, Object> conversion = keyExpression.getComparandConversionFunction();
-        if (conversion != null) {
-            return new QueryKeyExpressionWithComparison(keyExpression, new ConversionParameterComparison(type, param, conversion));
+        if (keyExpression.getComparandConversionFunction() != null) {
+            return new QueryKeyExpressionWithComparison(keyExpression, new ConversionParameterComparison(type, param, keyExpression));
         } else {
             return new QueryKeyExpressionWithComparison(keyExpression, new Comparisons.ParameterComparison(type, param));
         }
     }
 
-    private final class ConversionSimpleComparison extends Comparisons.SimpleComparison {
+    private static final class ConversionSimpleComparison extends Comparisons.SimpleComparisonBase {
```

Comment:
Should this maybe moved to `Comparisons`?

Reply:
I'm reluctant because of the `QueryKeyExpression` field, which would then sort-of create a circular dependency of concepts.

Reply:
Okay.